### PR TITLE
Fix cancel button in Edit features with feature-linked annotation

### PR DESCRIPTION
**Pull request by sclaridge**
This PR fixes a bug noticed by @hardikc07 during verification. Previously, when the dialog box 's cancel button was clicked, the feature was still selected.

Now when when the update button is clicked, `editAttributesDialog.showAndWait()` returns true, otherwise it returns null, which is stored in an Optional. When the dialog is created, the features are updated based on the value in the Optional.

### CODE REVIEW

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java
Comment (JonLavi) [inline editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java:72]: Perhaps it's possible to return `false` here. That way we can convert the type `Optional<Boolean>` to just `Boolean`.

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java
Comment (JonLavi) [inline editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java:232]: If the suggestion of returning just a `Boolean` from the dialog works, we can write here:
```suggestion
      if (updatingFeature) {
```

**Review by JonLavi — APPROVED** [review]
Very nice fix @sclaridge 
I've got one suggestion, which would be returning a `Boolean` instead of an `Optional<Boolean>` from the dialog. It could simplify things a tad more. If that doesn't work at all, however, what you have here works well too, so I'll approve so that you can get a second review asap!

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java
Comment (sclaridge) [reply editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java:72]: `Dialog.showAndWait()` returns an Optional, so I don't think I can use a `Boolean`.

Comment (JonLavi) [reply editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java:72]: Ah no worries then!

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java
Comment (Rachael-E) [inline editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java]: extra space needed between else and bracket 
```suggestion
      else {
```

**Review by Rachael-E — APPROVED** [review]
thanks @sclaridge this change looks good! Just a minor syntax thing to fix up (missing an extra space between else and a bracket) then good for you to go ahead and merge.

**Pull request merged**
